src/mesh/amr_loadbalance.cpp

```diff
@@ -751,7 +751,7 @@ void Mesh::RedistributeAndRefineMeshBlocks(ParameterInput *pin, ApplicationInput
       auto pmb = FindMeshBlock(on);
       for (auto &var : pmb->vars_cc_) {
         restriction_cache.RegisterRegionHost(
-            irestrict++, ProResInfo::GetInteriorRestrict(pmb, NeighborBlock(), var),
+            irestrict++, ProResInfo::GetInteriorRestrict(pmb.get(), NeighborBlock(), var),
```

Comment:
Would `FindMeshBlock` returning a pointer rather than a shared pointer be more in the spirit of this PR?